test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -254,20 +255,28 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
 
     it('reports unfinished top-level `await`', async () => {
       const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
-        '--no-warnings',
-        fixtures.path('es-modules/tla/unresolved.js'),
+        '--input-type=module',
+        '--eval',
+        `
+        await new Promise(() => {});
+        `,
```

Comment:
Adding `--input-type=module` makes this explicitly ESM. The point of the test is that the ambiguous `.js` file is parsed correctly, so we can't add a marker of explicitness like `--input-type` or `.mjs`.

```suggestion
        fixtures.path('es-modules/tla/unresolved.js'),
```

The `--no-warnings`, however, should go away. It was there previously to hide the experimental warning about syntax detection, but it was unintentionally also hiding the warning about unsettled top-level await, which we _should_ be testing for.